src/theory/quantifiers/ceg_instantiator.cpp

```diff
@@ -599,34 +635,35 @@ bool CegInstantiator::constructInstantiationInc(Node pv,
 }
 
 bool CegInstantiator::doAddInstantiation( std::vector< Node >& vars, std::vector< Node >& subs, std::vector< Node >& lemmas ) {
-  if (vars.size() > d_num_input_variables)
+  if (vars.size() > d_input_vars.size() || !d_var_order_index.empty())
   {
     Trace("cbqi-inst-debug") << "Reconstructing instantiations...." << std::endl;
     std::map< Node, Node > subs_map;
     for( unsigned i=0; i<subs.size(); i++ ){
       subs_map[vars[i]] = subs[i];
     }
     subs.clear();
-    for (unsigned i = 0; i < d_vars.size(); i++)
+    for (unsigned i = 0; i < d_input_vars.size(); i++)
```

Comment:
Avoid calling size() in each iteration. As discussed in todays meeting, etiher use iterators or the following pattern:

```
for (unsigned i = 0, size = d_input_vars.size(); i < size; ++i)
{
  ...
}
```

Reply:
If this is a matter of performance, I'm not convinced that there is an advantage to this pattern.  It seems there is an advantage in the fact that size is called less often, but this is at the cost of introducing a new unsigned on the stack.  I'd prefer doing some performance testing for this.  My guess is that it doesn't make a difference, but I'd like to be sure.

Reply:
Here is an excellent answer to that question:
[https://stackoverflow.com/a/3901666](https://stackoverflow.com/a/3901666)